### PR TITLE
handle non binary hmac sha key error

### PR DESCRIPTION
**Pull request by alecostard**
Hi! Did not know whether I should have opened an issue first, so feel free to close this if not applicable.

I've recently observed the following situation:

```
iex> Joken.Signer.create("HS256", nil, nil)

** (Joken.Error) Couldn't recognize the signer algorithm.

Possible values are:

["HS256", "HS384", "HS512", "RS256", "RS384", "RS512", "ES256", "ES384", "ES512", "PS256", "PS384", "PS512", "Ed25519", "Ed25519ph", "Ed448", "Ed448ph"]
```

which is confusing since the `"HS256"` is the first one in the list of possible values. The cause seems to be that when the algorithm is one of the HSXXX and the secret is not binary the only clause that matches is the unrecognized algorithm one.

The changes in this PR ensures that an unrecognized algorithm error only happens if the algorithm is indeed unrecognized. The offending code now raises a FunctionClauseError.

I thought about adding a test for this but since I saw no test for other malformed inputs (such as `create("HS256", "", nil)` raising a BadMapError), I decided to leave it as it is. 

Thanks for the library!

### CODE REVIEW

File: lib/joken/signer.ex [hunk 1]
```diff
@@ -99,7 +99,8 @@ defmodule Joken.Signer do
   def create(alg, _key, _headers) when alg in @map_key_algorithms,
     do: raise(Joken.Error, :algorithm_needs_key)
 
-  def create(_, _, _), do: raise(Joken.Error, :unrecognized_algorithm)
+  def create(alg, _, _) when alg not in @algorithms,
+    do: raise(Joken.Error, :unrecognized_algorithm)
```
Comment (victorolinasc) [inline lib/joken/signer.ex:103]: I think we should guard against anything other than binary and maps with a specific error type here. 

The way it is here it will blow with a pattern match error which would be cryptic too. 

Wdyt?

Comment (alecostard) [reply lib/joken/signer.ex:103]: Do you mean an additional clause with those guards on the key?

Comment (victorolinasc) [reply lib/joken/signer.ex:103]: I mean, before this PR we had a catch-all clause that would match on anything. With this here we do not have a catch all anymore.

Comment (alecostard) [reply lib/joken/signer.ex]: Ah, I see. I'll change the scope of the PR to address only the actual issue then.

**Review by victorolinasc — APPROVED** [review]
